src/spaces/compartment.jl

```diff
@@ -0,0 +1,188 @@
+export CompartmentSpace
+
+struct CompartmentSpace{D,F} <: AbstractSpace
+  grid::GridSpace
+  update_vel!::F
+  dims::NTuple{D, Int}
+  periodic::Bool
+  D::Int
```

Comment:
D as a field is not needed anymore, and it is anyway part of the type parameter of the type so it is accessible without being an explicit field.